HeroVault/HeroVault/Features/Game/GameScreenViewController.swift

```diff
@@ -0,0 +1,179 @@
+        CGSize(width: 170, height: 210)
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        guard let gameScreenCollectionViewCell =
+                gameCollectionView.dequeueReusableCell(withReuseIdentifier:
+                                                        Constants.SegueIdentifierNames.gameScreenCollectionViewCellName,
+                                                       for: indexPath) as? GameScreenCollectionViewCell else {
+            return UICollectionViewCell()
+        }
+        gameScreenCollectionViewCell.setUpNib(gameImage: gameScreenViewModel.populateCellImage)
+        return gameScreenCollectionViewCell
+    }
+    
+    private func handleCorrectAnswer(for cell: GameScreenCollectionViewCell) {
+        cell.correctAnswer()
+        reloadView()
+        currentScoreNumber()
+    }
+    
+    private func handleIncorrectAnswer(for cell: GameScreenCollectionViewCell, collectionView: UICollectionView) {
+        cell.incorrectAnswer()
+        gameScreenViewModel.cellsClickable = false
+        gameScreenViewModel.newHighScore() ? showNewHighScore() : showError()
+        gameScreenViewModel.resetScore()
+        highlightCorrectCell(in: collectionView)
+    }
+    
+    private func highlightCorrectCell(in collectionView: UICollectionView) {
+        guard let correctIndex = gameScreenViewModel.correctIndex else {
+            return
+        }
+        guard let correctCell = collectionView.cellForItem(at: IndexPath(row: correctIndex, section: 0)) as?
+                GameScreenCollectionViewCell else {
+            return
+        }
+        correctCell.correctAnswer()
+    }
+}
+
+extension GameScreenViewController: ViewModelProtocol {
+    func startLoadingIndicator() {
+        view.showLoadingIndicator()
+        gameCollectionView.isHidden = true
+        currentScoreLabel.isHidden = true
+        playButtonLabel.isHidden = true
+        highScoreLabel.isHidden = true
+        instructionLabel.isHidden = true
+    }
+    
+    func stopLoadingIndicator() {
+        view.stopLoadingIndicator()
+        gameCollectionView.isHidden = false
+        currentScoreLabel.isHidden = false
+        highScoreLabel.isHidden = false
+        instructionLabel.isHidden = false
+    }
+    
+    func reloadView() {
+        setUpView()
+        setUpCollectionView()
+        gameCollectionView.reloadData()
+    }
+    
+    func showError() {
+        let alert = UIAlertController(title: "Game Over", message:  "Your score was: \(gameScreenViewModel.score)", preferredStyle: .alert)
+        alert.addAction(UIAlertAction(title: "OK", style: .default) { _ in
+            self.playButtonLabel.isHidden = false
+            self.gameScreenViewModel.cellsClickable = false
+        })
+        self.present(alert, animated: true, completion: nil)
+    }
+    
+    func showNewHighScore() {
```

Comment:
and this will look like 
func showNewHighScore(title: title, message: message) {
showAlert(title: title, message: message) {
self.playButtonLabel.isHidden = false
self.gameScreenViewModel.cellsClickable = false
}
}